x-pack/plugins/data_enhanced/server/search/session/session_service.ts

```diff
@@ -368,15 +353,24 @@ export class SearchSessionService implements ISessionService {
     }
   };
 
+  public async getSearchIdMapping(deps: SearchSessionDependencies, sessionId: string) {
```

Comment:
Am I understanding correctly that this returns a mapping of each id to the strategy? Why do we need an API for that?

Reply:
One of the main changes in this PR is that I moved the `search` logic from the session service into the search service. The search service doesn't call `search` on the session service anymore, just `getId`, `trackId`, and `getSearchIdMapping` (as well as some of the session client methods).